gateway-sp-comms/src/single_sp.rs

```diff
@@ -836,6 +958,24 @@ fn sp_busy_policy() -> backoff::ExponentialBackoff {
     }
 }
 
+// Helper trait to provide methods on `io::Cursor` that are currently unstable.
+trait CursorUnstable {
```

Comment:
Nit: When I was reading the code above, I thought that `stable` was some type of property of the cursor itself. Maybe just rename this to `CursorHelper` or something so as not to give that impression.

Reply:
I'll go with `CursorExt` since it's sort of like an extension trait.